docs/testrunners.rst

```diff
@@ -0,0 +1,182 @@
+Test Runner
+===========
+
+In Cosmic Ray we use test runners to allow us to run any type of test
+that your project uses.
+
+Implementation details
+----------------------
+
+Cosmic Ray uses various packages that can be found under
+plugins/test-runners.
+
+Cosmic Ray comes with out of the box supported plugins for:
+1. unittest
+2. pytest
+3. nose
+4. nose2
+
+Each test runner is ultimately a subclass of
+``cosmic_ray.testing.TestRunner``.
+
+Each mutation will invoke another instance of the test runner, in order
+to run all tests for this specific mutant.
+
+Implementing a test runner
+--------------------------
+
+To implement a new test runner you need to create a new package.
+An example is provided at the end.
+
+The test runner must inherit TestRunner.
+Test Runner has 3 methods:
+1. __init__
+2. _run
+3. __call__
+
+__init__ is already implemented as part of TestRunner, and provides
+initialization for test_args.
+
+The abstract method is _run:
+It should discover all the tests that it should run and returns the results.
+The results are returned as a (success, result)
+tuple. `success` is a boolean indicating if the tests
+passed. `result` is any object that is appropriate to provide
+more information about the success/failure of the tests.
+
+Test Runners are provided as plugins
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cosmic Ray is designed to be extended with arbitrary test runners provided
+by users. It dynamically discovers tets runners at runtime using the
+``stevedore`` plugin system which relies on the ``setuptools``
+``entry_points`` concept. To make a new plugin available to Cosmic Ray
+you need to create a ``cosmic_ray.test_runners`` entry point; this is
+generally done in ``setup.py``. We'll show an example of how to do this
+later.
+
+A full example: ``AnotherRunner``
+---------------------------------
+
+One of the plugins bundled with Cosmic Ray is
+``cosmic_ray.testing.unittest_runner``.
+This test runner uses unittest package. It looks for test cases using
+the unittest package discover routine.
+To demonstrate how to create a test runner, we'll step through how to
+create that test runner in a new package called ``cosmic_ray_another_runner``.
+
+Creating the operator class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The initial layout for our package is like this:
+
+::
+
+    setup.py
+    cosmic_ray_another_runner/
+      __init__.py
+      version.py
```

Comment:
It looks like `version.py` isn't used below, so let's remove it. I think it's good to leave `version.py` out since it's not necessary and might confuse some people.